pages/api/mail.ts

```diff
@@ -0,0 +1,23 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { IResponse } from "../../interface/IResponse";
+import { generateHTML } from "../../helpers/generateHTML";
+import { sendMessageToUser } from "../../helpers/sendMessageToUser";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<IResponse>
+) {
+  const emailAddress: string = req.body.email;
```

Comment:
What about validation?
You can use class-validator to validate incoming data -> https://www.npmjs.com/package/class-validator
User could modify the request after it passes front-end validation